lib/mail_catcher/web/application.rb

```diff
@@ -5,18 +5,53 @@
 require "uri"
 
 require "sinatra"
-require "skinny"
-
-require "mail_catcher/bus"
+require 'async/websocket/adapters/rack'
 require "mail_catcher/mail"
 
-class Sinatra::Request
-  include Skinny::Helpers
-end
-
 module MailCatcher
   module Web
     class Application < Sinatra::Base
+      def initialize
+        super
+        @connections = Set.new
+        @semaphore = Async::Semaphore.new(512)
+
+        @count = 0
+      end
+
+      def connect(connection)
+        @connections << connection
+
+        @count += 1
+      end
+
+      def each(&block)
+        @connections.each(&block)
+      end
+
+      def broadcast(message)
+        Console.logger.info "Broadcast: #{message.inspect}"
```

Comment:
Can we consistently use either `Async.logger` or `Console.logger`? I think the former is more correct?

Reply:
Sure, I will use `Async.logger`